### PR TITLE
Implement Decodable for TransactionSignedEcRecovered

### LINKED ISSUE #1590: Implement `Decodable` for `TransactionSignedEcRecovered`
### Describe the feature

`Encodable` is already implemented for `TransactionSignedEcRecovered` . This PR implements `Decodable`.

I would like to use a recovered transaction within a custom struct and encode/decode the struct. This implementaton will allow me to derive `RlpDecodable` for my struct.

```
#[derive(RlpEncodable, RlpDecodable)]
struct MyStruct {
    inner: TransactionSignedEcRecovered,
}
```

I don't fully understand when `Header` is necessary for the serialization to be correct. I implemented decodable based on how the struct is encoded. If I'm missing something, please let me know so I can correct this mistake. Thanks for considering this issue.

### Additional context

I am using reth as a library to build a layer 1 blockchain. I realize that this is not in scope right now (based on feedback from another PR I opened #1281 ) but I look forward to making more contributions when these features are within scope.

### PR DESCRIPTION
**Pull request by grantkee**
## Changes
- implement `Decodable` for `TransactionSignedEcRecovered`
- all tests pass
- `cargo +nightly fmt -- --check` required moving `H256, U256` to a new line
- no clippy warnings

## Question
- [ ] Would you prefer a new `DecodeError` or is `Custom` acceptable here?

closes #1590

### CODE REVIEW

File: crates/primitives/src/transaction/mod.rs [hunk 1]
```diff
@@ -899,6 +899,16 @@ impl Encodable for TransactionSignedEcRecovered {
     }
 }
 
+impl Decodable for TransactionSignedEcRecovered {
+    fn decode(buf: &mut &[u8]) -> Result<Self, DecodeError> {
+        let signed_transaction = TransactionSigned::decode(buf)?;
+        let signer = signed_transaction
+            .recover_signer()
+            .ok_or(DecodeError::Custom("Unable to recover decoded transaction signer."))?;
```
Comment (mattsse) [inline crates/primitives/src/transaction/mod.rs:907]: I guess this is fine, since this is basically just a convenience impl

wdyt @Rjected

**Review by mattsse — CHANGES_REQUESTED** [review]
please restore lock file

**Comment by grantkee** [comment]
I restored the Cargo.lock

File: crates/primitives/src/transaction/mod.rs (hunk 1 above)
Comment (Rjected) [reply crates/primitives/src/transaction/mod.rs:907]: yeah, this seems alright as a convenience impl

**Review by mattsse — APPROVED** [review]

**Pull request merged**